### ISSUE TITLE
Add support for Windows

### ISSUE THREAD

**Issue by musicman3320** [musicman3320/m2r#24]
Code to update: 
Locating M2 path
Launching M2 just before socket creation

**Comment by musicman3320** [comment]
@dkahle can you still look into this?  I am not familiar with how M2 works on Windows.

**Comment by musicman3320** [comment]
Do we still want to do this for the first release?  If it's too much work, I have no problem pushing it to a future version (unless, of course, CRAN won't accept it without complete cross-platform-ness).

**Comment by dkahle** [comment]
This is now very close, but I can't get M2 to run from DOS through Cygwin.  Installing Cygwin and M2 through the instructions on their website is easy, and opening Cygwin and running M2 is just as easy. The problem I'm having is starting the M2 process from DOS.
```dos
cmd.exe /K C:/cygwin/bin/env.exe C:/cygwin/bin/M2.exe
```
I get the error that it can't find lapack DLL, which is actually in `C:/cygwin/lib`. Thoughts?

**Comment by musicman3320** [comment]
Is there a command line parameter to env.exe to specify library locations?  

Also, how do you run M2 in Cygwin successfully?

**Comment by dkahle** [comment]
Hm, good question about env.exe. I'll check.

When you run Cygwin (even from DOS), it essentially opens a bash shell.  From there, just open M2 with `M2` and carry on.

**Comment by musicman3320** [comment]
That's what I thought.  My guess is env.exe does some initialization when run from the command prompt, which is being bypassed when running a command directly.  

At worst, we could handle the process a bit differently on Windows, essentially piping in the M2 code to STDIN for the command prompt.  Avoiding this would be nice, but at least it would still work as expected, since once the M2 server is running everything functions normally.

**Comment by dkahle** [comment]
So I've got it running, but it crashes on startup. I think it'll just take tinkering. Do you have a Windows machine available?

**Comment by sommars** [comment]
I have a Windows machine I can test on.

**Comment by dkahle** [comment]
Ok, pull down and test on Windows.  I've documented some comments in `do_start_m2()`.  My recommendation is to create a new RStudio project on your Windows machine checking out the repo here, pull that down, load it, then try to run that function...

**Comment by sommars** [comment]
Gosh, I'm struggling just to get M2 to run on a Windows machine

**Comment by dkahle** [comment]
Just follow the directions explicitly.  Use the 32-bit version, you should be fine.  We can Skype later if need be.  (Unfortunately, if you're struggling... that doesn't bode well for the average user.)

**Comment by sommars** [comment]
It's probably my fault. I've never used cygwin before.

**Comment by musicman3320** [comment]
Robert and I have tried several things with m2r on Windows.  We have confirmed the following.  

- The M2 session in Cygwin launches and runs as expected.  In particular, if R launches an M2 session with code that performs a few small computations and exits, the M2 process launched by R ends as quickly as expected, whereas if R launch M2 with code that should hang (say, an infinite loop) the process remains open until we kill it.  
- Socket servers created by M2 in Cygwin can be connected to by other M2 instances in Cygwin, and by telnet in Windows.  In particular, the `openInOut` function works as expected in Cygwin.  
- Socket servers created by telnet can be connected to by R via the `socket()` function.  
- Despite the above, M2 and R are unable to connect to one another.  It remains unclear why this is, since both interfaces work as expected when connecting to (for instance) telnet.  

One possible (though clearly cheating) workaround: launch a telnet session to act as a "middle man" of sorts.  I really dislike this solution, but we do what we have to, I guess.
